pkg/cmd/util/cmd.go

```diff
@@ -31,3 +33,32 @@ func GetDisplayFilename(filename string) string {
 
 	return filename
 }
+
+// ResolveResourceSyntax will return the kind (if found), name and a possible error
+// from the provided arg. If a default kind has been specified, then arg
+// should respect it.
+func ResolveResourceSyntax(mapper meta.RESTMapper, defaultKind, arg string) (string, string, error) {
```

Comment:
I want this kind of resolution upstream too.  This is would allow resolution of types for logs without retrieving resources I don't care about.